src/main/java/seedu/contax/logic/commands/EditAppointmentCommand.java

```diff
@@ -0,0 +1,274 @@
+package seedu.contax.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_DURATION;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_PERSON;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_TIME;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.contax.commons.core.Messages;
+import seedu.contax.commons.core.index.Index;
+import seedu.contax.commons.util.CollectionUtil;
+import seedu.contax.commons.util.DateUtil;
+import seedu.contax.logic.commands.exceptions.CommandException;
+import seedu.contax.model.Model;
+import seedu.contax.model.appointment.Appointment;
+import seedu.contax.model.appointment.Duration;
+import seedu.contax.model.appointment.Name;
+import seedu.contax.model.appointment.StartDateTime;
+import seedu.contax.model.appointment.exceptions.OverlappingAppointmentException;
+import seedu.contax.model.person.Person;
+
+/**
+ * Edits the details of an existing appointment in the schedule.
+ */
+public class EditAppointmentCommand extends Command {
+
+    public static final String COMMAND_WORD = "editAppointment";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the appointment identified "
+            + "by the index number used in the displayed appointment list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_TIME + "TIME] "
+            + "[" + PREFIX_DURATION + "DURATION] "
+            + "[" + PREFIX_PERSON + "PERSON]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Product Meeting with John "
+            + PREFIX_DURATION + "60 "
+            + PREFIX_PERSON + "2";
+
+    public static final String MESSAGE_EDIT_APPOINTMENT_SUCCESS = "Edited Appointment: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "No fields supplied, nothing was changed.";
+    public static final String MESSAGE_OVERLAPPING_APPOINTMENT = "The new appointment will overlap with"
+            + " another appointment in the schedule!";
+
+    private final Index index;
+    private final EditAppointmentDescriptor editAppointmentDescriptor;
+
+    /**
```

Comment:
missed a javadoc header here

Reply:
Thanks for spotting it, fixed in commit 3d9af4e.